junit-runtime/src/main/scala/org/scalajs/junit/JUnitExecuteTest.scala

```diff
@@ -64,116 +64,101 @@ final class JUnitExecuteTest(taskDef: TaskDef, runner: JUnitBaseRunner,
+          val selector = new NestedTestSelector(fullyQualifiedName, methodName)
+          eventHandler.handle(new JUnitEvent(taskDef, Status.Failure, selector))
       }
     }
-  }
 
-  private[this] def executedWithoutExceptions(methodName: String,
-    testAnnotation: org.junit.Test, timeInSeconds: Double) = {
-    if (testAnnotation.expected == classOf[org.junit.Test.None]) {
-      if (runner.runSettings.verbose)
-        logFormattedInfo(methodName, "started")
-      taskPassed(methodName)
-    } else {
-      val msg = {
-        s"failed: Expected exception: ${testAnnotation.expected} " +
-        s"took $timeInSeconds sec"
+    try {
+      for (method <- jUnitMetadata.beforeMethod) {
+        if (runner.runSettings.verbose)
+          logFormattedInfo(methodName, "started")
+        classMetadata.invoke(testClassInstance, methodName)
       }
-      logFormattedError(methodName, msg, None)
-      taskFailed(methodName)
-    }
-    runner.taskRegisterTotal()
-  }
+      executeTestMethods()
+    } catch {
+      case ex: AssumptionViolatedException =>
+        logAssertionWarning(methodName, ex, timeInSeconds)
 
-  private[this] def executedWithExceptions(methodName: String,
-      testAnnotation: org.junit.Test, timeInSeconds: Double, ex: Throwable) = {
-    if (classOf[AssumptionViolatedException].isInstance(ex) ||
-        classOf[internal.AssumptionViolatedException].isInstance(ex)) {
-      logFormattedInfo(methodName, "started")
-      logAssertionWarning(methodName, ex, timeInSeconds)
-      taskSkipped()
-    } else if (testAnnotation.expected.isInstance(ex)) {
-      if (runner.runSettings.verbose)
-        logFormattedInfo(methodName, "started")
-      taskPassed(methodName)
-    } else if (testAnnotation.expected == classOf[org.junit.Test.None]) {
-      val failedMsg = new StringBuilder
-      failedMsg ++= "failed: "
-      if (ex.isInstanceOf[AssertionError] && runner.runSettings.logAssert) {
-        failedMsg ++= "java.lang." ++= c("AssertionError", ERRMSG) ++= ": "
-        failedMsg ++= ex.getMessage
-      } else if (runner.runSettings.logExceptionClass) {
-        failedMsg ++= ex.getMessage
-      } else {
-        failedMsg ++= ex.getClass.toString ++= " expected<"
-        failedMsg ++= testAnnotation.expected.toString ++= "> but was<"
-        failedMsg ++= ex.getClass.toString += '>'
-      }
-      failedMsg += ','
-      val msg = s"$failedMsg took $timeInSeconds sec"
-      val exOpt = {
-        if (!ex.isInstanceOf[AssertionError] || runner.runSettings.logAssert) Some(ex)
-        else None
-      }
-      logFormattedError(methodName, msg, exOpt)
-      taskFailed(methodName)
-    } else {
-      val msg = s"failed: ${ex.getClass}, took $timeInSeconds sec"
-      logFormattedError(methodName, msg, Some(ex))
-      taskFailed(methodName)
+      case ex: internal.AssumptionViolatedException =>
+        logAssertionWarning(methodName, ex, timeInSeconds)
```

Comment:
What about other exceptions? (I realize this is not changing the existing behavior, but it seems odd, since it can cause `executeTests` to throw).